src/main/java/com/contentful/java/cma/ModuleUploads.java

```diff
@@ -95,48 +119,70 @@ public ModuleUploads(Retrofit retrofit, Executor callbackExecutor) {
    * @param spaceId a nonnull id representing the space to add the upload to.
    * @param stream  the actual binary representation of the upload. Cannot be null.
    * @return the upload created, containing the id to be used further on.
-   * @throws IllegalArgumentException if spaceId is null.
-   * @throws IllegalArgumentException if stream is null.
-   * @throws java.io.IOException      if the stream could not be read.
+   * @throws IllegalArgumentException        if spaceId is null.
+   * @throws IllegalArgumentException        if stream is null.
+   * @throws java.io.IOException             if the stream could not be read.
+   * @throws CMANotWithEnvironmentsException if environmentId was set using
+   *                                         {@link CMAClient.Builder#setEnvironmentId(String)}
+   *                                         throws a runtime exception.
    */
   public CMAUpload create(String spaceId, InputStream stream) throws IOException {
     assertNotNull(spaceId, "spaceId");
     assertNotNull(stream, "stream");
+    throwIfEnvironmentIdIsNotDefault();
```

Comment:
This should not throw an error, as updates are environment agnostic and any upload can be used from any environment, and it's a VERY useful thing to have.